app/src/main/java/de/tum/in/tumcampusapp/component/tumui/person/PersonDetailsActivity.kt

```diff
@@ -0,0 +1,210 @@
+
+        // Setup employee groups
+        val groups = employee.groups
+        if (groups?.isNotEmpty() == true) {
+            groupsRecyclerView.setHasFixedSize(true)
+            groupsRecyclerView.layoutManager = LinearLayoutManager(this)
+            groupsRecyclerView.adapter = PersonGroupsAdapter(groups)
+        } else {
+            dividerNameGroups.visibility = View.GONE
+            groupsRecyclerView.visibility = View.GONE
+        }
+
+        // Setup contact items
+        val contactItems = arrayListOf<AbstractContactItem>().apply {
+            add(EmailContactItem(employee.email))
+            employee.businessContact?.let { contact ->
+                if (contact.homepage.isNotBlank()) {
+                    add(HomepageContactItem(contact.homepage))
+                }
+            }
+        }
+
+        employee.telSubstations?.forEach { station ->
+            if (station.number.isNotBlank()) {
+                contactItems.add(PhoneContactItem(station.number))
+            }
+        }
+
+        employee.businessContact?.let { contact ->
+            if (contact.mobilephone.isNotBlank()) {
+                contactItems.add(MobilePhoneContactItem(contact.mobilephone))
+            }
+
+            if (contact.additionalInfo.isNotBlank()) {
+                contactItems.add(InformationContactItem(contact.additionalInfo))
+            }
+        }
+
+        if (employee.consultationHours.isNotBlank()) {
+            contactItems.add(OfficeHoursContactItem(employee.consultationHours))
+        }
+
+        employee.rooms?.let { rooms ->
+            rooms.forEach { room ->
+                contactItems.add(RoomContactItem(room.getFullLocation(), room.number))
+            }
+        }
+
+        if (contactItems.isNotEmpty()) {
+            contactItemsRecyclerView.setHasFixedSize(true)
+            contactItemsRecyclerView.layoutManager = LinearLayoutManager(this)
+            contactItemsRecyclerView.adapter = PersonContactItemsAdapter(contactItems)
+        } else {
+            dividerGroupsContactItems.visibility = View.GONE
+            contactItemsRecyclerView.visibility = View.GONE
+        }
+    }
+
+    /**
+     * Adds the given employee to the users contact list
+     *
+     * @param employee Object to insert into contacts
+     */
+    private fun addContact(employee: Employee?) {
+        if (!isPermissionGranted(0)) {
+            return
+        }
+
+        if (employee != null) {
+            ContactsManager.saveToContacts(this, employee)
+        }
+    }
+
+    /**
+     * Check Contacts permission for Android 6.0
+     *
+     * @param id the request id
+     * @return If the contacts permission was granted
+     */
+    private fun isPermissionGranted(id: Int): Boolean {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_CONTACTS) == PackageManager.PERMISSION_GRANTED) {
+            return true
+        } else {
```

Comment:
You don't need an `else` here, as the first part of the condition returns in case it is true.

Reply:
Fixed